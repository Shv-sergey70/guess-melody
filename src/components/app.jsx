import React from 'react';
import WelcomeScreen from './welcome-screen';

const App = () => {
  return (
    <WelcomeScreen/>
  );
};

export default App;
