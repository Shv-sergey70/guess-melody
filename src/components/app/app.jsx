import React from 'react';
import PropTypes from 'prop-types';
import WelcomeScreen from '../welcome-screen/welcome-screen';
import ArtistQuestionScreen from "../artist-question-screen/artist-question-screen";
import GenreQuestionScreen from "../genre-question-screen/genre-question-screen";
import {connect} from "react-redux";

const App = ({currentStep, questions, time, attempts}) => {
  if (currentStep === -1) {
    return (
      <WelcomeScreen time={time} attempts={attempts} />
    );
  }

  switch (questions[currentStep].type) {
    case `genre`:
      return (
        <GenreQuestionScreen
          question={questions[currentStep]}
          attempts={attempts} />
      );
    case `artist`:
      return (
        <ArtistQuestionScreen
          question={questions[currentStep]}
          attempts={attempts} />
      );
  }

  return null;
};

App.propTypes = {
  questions: PropTypes.arrayOf(PropTypes.object),
  time: PropTypes.number.isRequired,
  attempts: PropTypes.number.isRequired,
  currentStep: PropTypes.number.isRequired
};

const mapStateToProps = ({step}) => ({
  currentStep: step
});

export {App};
export default connect(mapStateToProps)(App);
