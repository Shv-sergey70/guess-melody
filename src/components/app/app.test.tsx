import * as React from 'react';
import {configure, mount} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {App} from "./app";
import {MemoryRouter} from 'react-router-dom';

configure({adapter: new Adapter()});

jest.mock(`../../hocs/with-active-player/with-active-player.jsx`);
jest.mock(`../question-screen-layout/question-screen-layout.jsx`, () => `question-screen-layout`);
jest.mock(`../welcome-screen/welcome-screen.jsx`, () => `welcome-screen`);
jest.mock(`../authorization-screen/authorization-screen.jsx`, () => `authorization-screen`);
jest.mock(`../losing-screen/losing-screen.jsx`, () => `losing-screen`);

const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `./tracks/Anivar_Summer.mp3`,
        genre: `pop`
      },
      {
        src: `./tracks/Billie_Eilish_ilomilo.mp3`,
        genre: `jazz`
      },
      {
        src: `./tracks/Cj_L_Black_Rain.mp3`,
        genre: `rock`
      },
      {
        src: `./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3`,
        genre: `classic`
      }
    ]
  },
  {
    type: `artist`,
    song: {
      artist: `Artur Latte`,
      src: `./tracks/Raim_Artur_Latte.mp3`,
    },
    answers: [
      {
        picture: `/img/default-artist.png`,
        artist: `Billie Eilish`
      },
      {
        picture: `/img/default-artist.png`,
        artist: `Imagine Dragons`
      },
      {
        picture: `/img/default-artist.png`,
        artist: `Artur Latte`
      }
    ]
  }
];

describe(`App correctly renders`, () => {
  test(`Welcome screen`, () => {
    const tree = mount(
        <MemoryRouter initialEntries={[`/`]} keyLength={0} >
          <App
            attempts={3}
            questions={questions}
            user={{}}
            currentStep={-1}
            isNoMoreTime={false}
            isNoMoreAttempts={false} />
        </MemoryRouter>
    );

    expect(tree.find(`App`)).toMatchSnapshot();
  });

  test(`Question-question screen`, () => {
    const tree = mount(
        <MemoryRouter initialEntries={[`/`]} keyLength={0} >
          <App
            attempts={3}
            questions={questions}
            user={{}}
            currentStep={1}
            isNoMoreTime={false}
            isNoMoreAttempts={false} />
        </MemoryRouter>
    );

    expect(tree.find(`App`)).toMatchSnapshot();
  });

  test(`Losing screen`, () => {
    const tree = mount(
        <MemoryRouter initialEntries={[`/lose`]} keyLength={0} >
          <App
            attempts={3}
            questions={questions}
            user={{}}
            currentStep={1}
            isNoMoreTime={true}
            isNoMoreAttempts={false} />
        </MemoryRouter>
    );

    expect(tree.find(`App`)).toMatchSnapshot();
  });

  test(`Authorization is required`, () => {
    const tree = mount(
        <MemoryRouter initialEntries={[`/auth`]} keyLength={0} >
          <App
            attempts={3}
            questions={questions}
            user={{}}
            currentStep={0}
            isNoMoreTime={false}
            isNoMoreAttempts={false} />
        </MemoryRouter>
    );

    expect(tree.find(`App`)).toMatchSnapshot();
  });
});
