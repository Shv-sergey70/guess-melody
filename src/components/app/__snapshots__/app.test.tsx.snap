// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders Authorization is required 1`] = `
<App
  attempts={3}
  currentStep={0}
  isNoMoreAttempts={false}
  isNoMoreTime={false}
  questions={
    Array [
      Object {
        "answers": Array [
          Object {
            "genre": "pop",
            "src": "./tracks/Anivar_Summer.mp3",
          },
          Object {
            "genre": "jazz",
            "src": "./tracks/Billie_Eilish_ilomilo.mp3",
          },
          Object {
            "genre": "rock",
            "src": "./tracks/Cj_L_Black_Rain.mp3",
          },
          Object {
            "genre": "classic",
            "src": "./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3",
          },
        ],
        "genre": "rock",
        "type": "genre",
      },
      Object {
        "answers": Array [
          Object {
            "artist": "Billie Eilish",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Imagine Dragons",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Artur Latte",
            "picture": "/img/default-artist.png",
          },
        ],
        "song": Object {
          "artist": "Artur Latte",
          "src": "./tracks/Raim_Artur_Latte.mp3",
        },
        "type": "artist",
      },
    ]
  }
  user={Object {}}
>
  <Switch>
    <Route
      component={[Function]}
      computedMatch={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/auth",
          "url": "/auth",
        }
      }
      exact={true}
      location={
        Object {
          "hash": "",
          "pathname": "/auth",
          "search": "",
          "state": undefined,
        }
      }
      path="/auth"
    >
      <WithLogin
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/auth",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/auth",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/auth",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/auth",
            "url": "/auth",
          }
        }
      >
        <authorization-screen
          email=""
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/auth",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/auth",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isDisabled={true}
          location={
            Object {
              "hash": "",
              "pathname": "/auth",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/auth",
              "url": "/auth",
            }
          }
          onEmailChange={[Function]}
          onPasswordChange={[Function]}
          password=""
        />
      </WithLogin>
    </Route>
  </Switch>
</App>
`;

exports[`App correctly renders Losing screen 1`] = `
<App
  attempts={3}
  currentStep={1}
  isNoMoreAttempts={false}
  isNoMoreTime={true}
  questions={
    Array [
      Object {
        "answers": Array [
          Object {
            "genre": "pop",
            "src": "./tracks/Anivar_Summer.mp3",
          },
          Object {
            "genre": "jazz",
            "src": "./tracks/Billie_Eilish_ilomilo.mp3",
          },
          Object {
            "genre": "rock",
            "src": "./tracks/Cj_L_Black_Rain.mp3",
          },
          Object {
            "genre": "classic",
            "src": "./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3",
          },
        ],
        "genre": "rock",
        "type": "genre",
      },
      Object {
        "answers": Array [
          Object {
            "artist": "Billie Eilish",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Imagine Dragons",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Artur Latte",
            "picture": "/img/default-artist.png",
          },
        ],
        "song": Object {
          "artist": "Artur Latte",
          "src": "./tracks/Raim_Artur_Latte.mp3",
        },
        "type": "artist",
      },
    ]
  }
  user={Object {}}
>
  <Switch>
    <Route
      component="losing-screen"
      computedMatch={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/lose",
          "url": "/lose",
        }
      }
      exact={true}
      location={
        Object {
          "hash": "",
          "pathname": "/lose",
          "search": "",
          "state": undefined,
        }
      }
      path="/lose"
    >
      <losing-screen
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/lose",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/lose",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/lose",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/lose",
            "url": "/lose",
          }
        }
      />
    </Route>
  </Switch>
</App>
`;

exports[`App correctly renders Question screen 1`] = `
<App
  attempts={3}
  currentStep={1}
  isNoMoreAttempts={false}
  isNoMoreTime={false}
  questions={
    Array [
      Object {
        "answers": Array [
          Object {
            "genre": "pop",
            "src": "./tracks/Anivar_Summer.mp3",
          },
          Object {
            "genre": "jazz",
            "src": "./tracks/Billie_Eilish_ilomilo.mp3",
          },
          Object {
            "genre": "rock",
            "src": "./tracks/Cj_L_Black_Rain.mp3",
          },
          Object {
            "genre": "classic",
            "src": "./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3",
          },
        ],
        "genre": "rock",
        "type": "genre",
      },
      Object {
        "answers": Array [
          Object {
            "artist": "Billie Eilish",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Imagine Dragons",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Artur Latte",
            "picture": "/img/default-artist.png",
          },
        ],
        "song": Object {
          "artist": "Artur Latte",
          "src": "./tracks/Raim_Artur_Latte.mp3",
        },
        "type": "artist",
      },
    ]
  }
  user={Object {}}
>
  <Switch>
    <Route
      computedMatch={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
      location={
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      path="/"
      render={[Function]}
    >
      <question-screen-layout
        question={
          Object {
            "answers": Array [
              Object {
                "artist": "Billie Eilish",
                "picture": "/img/default-artist.png",
              },
              Object {
                "artist": "Imagine Dragons",
                "picture": "/img/default-artist.png",
              },
              Object {
                "artist": "Artur Latte",
                "picture": "/img/default-artist.png",
              },
            ],
            "song": Object {
              "artist": "Artur Latte",
              "src": "./tracks/Raim_Artur_Latte.mp3",
            },
            "type": "artist",
          }
        }
      />
    </Route>
  </Switch>
</App>
`;

exports[`App correctly renders Welcome screen 1`] = `
<App
  attempts={3}
  currentStep={-1}
  isNoMoreAttempts={false}
  isNoMoreTime={false}
  questions={
    Array [
      Object {
        "answers": Array [
          Object {
            "genre": "pop",
            "src": "./tracks/Anivar_Summer.mp3",
          },
          Object {
            "genre": "jazz",
            "src": "./tracks/Billie_Eilish_ilomilo.mp3",
          },
          Object {
            "genre": "rock",
            "src": "./tracks/Cj_L_Black_Rain.mp3",
          },
          Object {
            "genre": "classic",
            "src": "./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3",
          },
        ],
        "genre": "rock",
        "type": "genre",
      },
      Object {
        "answers": Array [
          Object {
            "artist": "Billie Eilish",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Imagine Dragons",
            "picture": "/img/default-artist.png",
          },
          Object {
            "artist": "Artur Latte",
            "picture": "/img/default-artist.png",
          },
        ],
        "song": Object {
          "artist": "Artur Latte",
          "src": "./tracks/Raim_Artur_Latte.mp3",
        },
        "type": "artist",
      },
    ]
  }
  user={Object {}}
>
  <Switch>
    <Route
      computedMatch={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
      location={
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      path="/"
      render={[Function]}
    >
      <welcome-screen
        attempts={3}
      />
    </Route>
  </Switch>
</App>
`;
